presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3ClientFactory.java

```diff
@@ -164,14 +157,133 @@ private static AWSCredentialsProvider getCustomAWSCredentialsProvider(Configurat
+                                .numRetries(maxErrorRetries))
+                        .apiCallTimeout(java.time.Duration.ofMillis(socketTimeout.toMillis()))
+                        .apiCallAttemptTimeout(java.time.Duration.ofMillis(connectTimeout.toMillis()))
+                        .putAdvancedOption(SdkAdvancedClientOption.USER_AGENT_PREFIX, userAgentPrefix)
+                        .putAdvancedOption(SdkAdvancedClientOption.USER_AGENT_SUFFIX, s3UserAgentSuffix))
+                .httpClientBuilder(NettyNioAsyncHttpClient.builder()
+                        .maxConcurrency(maxConnections)
+                        .connectionTimeout(java.time.Duration.ofMillis(connectTimeout.toMillis()))
+                        .readTimeout(java.time.Duration.ofMillis(socketTimeout.toMillis())));
+
+        configureRegionAndEndpoint(clientBuilder, config, defaults);
+        return clientBuilder.build();
+    }
+
+    private void configureRegionAndEndpoint(S3ClientBuilder clientBuilder, Configuration config, HiveS3Config defaults)
+    {
+        boolean regionOrEndpointSet = false;
+
+        String endpoint = config.get(S3_ENDPOINT);
+        boolean pinS3ClientToCurrentRegion = config.getBoolean(S3_PIN_CLIENT_TO_CURRENT_REGION, defaults.isPinS3ClientToCurrentRegion());
+        verify(!pinS3ClientToCurrentRegion || endpoint == null,
+                "Invalid configuration: either endpoint can be set or S3 client can be pinned to the current region");
+
+        // use local region when running inside of EC2
+        if (pinS3ClientToCurrentRegion) {
+            try {
+                Region region = software.amazon.awssdk.regions.providers.DefaultAwsRegionProviderChain.builder().build().getRegion();
+                if (region != null) {
+                    clientBuilder.region(region);
+                    regionOrEndpointSet = true;
+                    log.debug("Using region from provider chain: %s", region);
+                }
+            }
+            catch (Exception e) {
+                log.debug("Could not determine current region from provider chain: %s", e.getMessage());
+            }
+        }
+
+        if (!isNullOrEmpty(endpoint)) {
+            clientBuilder.endpointOverride(URI.create(endpoint));
+            log.debug("Using custom endpoint: %s", endpoint);
+            if (!regionOrEndpointSet) {
+                clientBuilder.region(Region.US_EAST_1);
+                log.debug("Setting default region US_EAST_1 for custom endpoint");
+            }
+            regionOrEndpointSet = true;
+        }
+
+        clientBuilder.forcePathStyle(true);
+
+        if (!regionOrEndpointSet) {
+            clientBuilder.region(Region.US_EAST_1);
+            log.debug("No region or endpoint specified, defaulting to US_EAST_1");
+        }
+    }
+
+    private void configureRegionAndEndpoint(S3AsyncClientBuilder clientBuilder, Configuration config, HiveS3Config defaults)
```

Comment:
We won't need the above method as we don't need the sync client, so please apply those changes to this method.